src/lib/viewers/ResultsViewer.ts

```diff
@@ -0,0 +1,136 @@
+import {RunResults, SeverityLevel, Violation} from '@salesforce/code-analyzer-core';
+import {Display} from '../Display';
+import {BundleName, getMessage} from '../messages';
+
+export interface ResultsViewer {
+	view(results: RunResults): void;
+}
+
+abstract class AbstractResultsViewer implements ResultsViewer {
+	protected display: Display;
+
+	public constructor(display: Display) {
+		this.display = display;
+	}
+
+	public abstract view(results: RunResults): void;
+}
+
+export class ResultsDetailViewer extends AbstractResultsViewer {
+	/**
+	 * The display format consumes a lot of vertical space. As such, we're choosing to limit the total
+	 * number of violations that it will display before truncating the results and instructing the user
+	 * to consult one of the output files.
+	 */
+	public static readonly RESULTS_CUTOFF = 10;
+
+	public view(results: RunResults): void {
+		if (results.getViolationCount() === 0) {
+			this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.found-no-results'));
+		} else {
+			const violations = this.sortViolations(results.getViolations());
+
+			this.displaySummary(violations);
+			this.displayDetails(violations);
+			this.displayBreakdown(results);
+		}
+	}
+
+	private countUniqueFiles(violations: Violation[]): number {
+		const fileSet: Set<string> = new Set();
+		violations.forEach(v => {
+			const primaryLocation = v.getCodeLocations()[v.getPrimaryLocationIndex()];
+			const file = primaryLocation.getFile();
+			if (typeof file === 'string') {
+				fileSet.add(file);
+			}
+		});
+		return fileSet.size;
+	}
+
+	private sortViolations(violations: Violation[]): Violation[] {
+		return violations.toSorted((v1, v2) => {
+			// First compare severities.
+			const v1Sev = v1.getRule().getSeverityLevel();
+			const v2Sev = v2.getRule().getSeverityLevel();
+			if (v1Sev !== v2Sev) {
+				return v1Sev < v2Sev ? -1 : 1;
```

Comment:
I'm not sure how strict the output of the sort method must be (if it looks for -1 and 1 exactly) but if it just looks for positive vs negative... then return v1Sev - v2Sev should be sufficient.

Reply:
Good catch. I keep forgetting that the Enums are technically numbers that you can do math with.

Reply:
(And yeah, AFAIK the sorting method just cares about the return being positive or negative, not -1 or 1 specifically.)